lib/Doctrine/ORM/QueryBuilder.php

```diff
@@ -740,21 +739,19 @@ public function add($dqlPartName, $dqlPart, $append = false)
      *         ->leftJoin('u.Phonenumbers', 'p');
      * </code>
      *
-     * @param mixed $select The selection expressions.
+     * @param mixed ...$select The selection expressions.
      *
      * @return static
      */
-    public function select($select = null)
+    public function select(...$select)
     {
         $this->_type = self::SELECT;
 
-        if (empty($select)) {
+        if ($select === []) {
             return $this;
         }
 
-        $selects = is_array($select) ? $select : func_get_args();
-
-        return $this->add('select', new Expr\Select($selects), false);
+        return $this->add('select', new Expr\Select($select), false);
```

Comment:
Hi, are you sure you haven't done a mistake here? I'm seeing a bug related to this place: https://github.com/doctrine/orm/issues/11241

Reply:
Possible, I'll take a look.